WordPress/Classes/Utility/ImmuTable.swift

```diff
@@ -0,0 +1,206 @@
+import Foundation
+import UIKit
+
+public typealias ImmuTableActionType = (ImmuTableRow) -> Void
+
+public protocol Reusable {
+    static var reusableIdentifier: String { get }
+}
+
+extension Reusable {
+    var reusableIdentifier: String {
+        get {
+            return self.dynamicType.reusableIdentifier
+        }
+    }
+}
+
+public protocol ImmuTableRow: Reusable {
+    var action: ImmuTableActionType? { get }
+    static var cellClass: AnyClass { get }
+    func configureCell(cell: UITableViewCell)
+    static var registrable: CellRegistrable { get }
+    static var customHeight: Float? { get }
+}
+
+public struct ImmuTableSection {
+    let headerText: String?
+    let rows: [ImmuTableRow]
+    let footerText: String?
+
+    init(rows: [ImmuTableRow]) {
+        self.headerText = nil
+        self.rows = rows
+        self.footerText = nil
+    }
+
+    init(headerText: String?, rows: [ImmuTableRow], footerText: String?) {
+        self.headerText = headerText
+        self.rows = rows
+        self.footerText = footerText
+    }
+}
+
+public enum CellRegistrable {
+    case Nib(UINib)
+    case Class(AnyClass)
+}
+
+public protocol CellRegistrator {
+    func register(registrable: CellRegistrable, cellReuseIdentifier: String)
+}
+
+
+extension UITableView: CellRegistrator {
+    public func register(registrable: CellRegistrable, cellReuseIdentifier: String) {
+        switch registrable {
+        case .Nib(let nib):
+            registerNib(nib, forCellReuseIdentifier: cellReuseIdentifier)
+        case .Class(let cellClass):
+            registerClass(cellClass, forCellReuseIdentifier: cellReuseIdentifier)
+        }
+    }
+}
+
+public struct ImmuTable {
+    public let sections: [ImmuTableSection]
+
+    public func rowAtIndexPath(indexPath: NSIndexPath) -> ImmuTableRow {
+        return sections[indexPath.section].rows[indexPath.row]
+    }
+
+    public static func registerRows(rows: [ImmuTableRow.Type], tableView: CellRegistrator) {
+        let registrables = rows.reduce([:]) {
+            (var classes, row) -> [String: CellRegistrable] in
+
+            classes[row.reusableIdentifier] = row.registrable
+            return classes
+        }
+        for (identifier, registrable) in registrables {
```

Comment:
I thought about it for a while. I think it should work, as the docs say

> If you previously registered a class or nib file with the same reuse identifier, the class you specify in the cellClass parameter replaces the old entry. You may specify nil for cellClass if you want to unregister the class from the specified reuse identifier.

However, I feel like that's relying too much on an implementation detail. I just feel safer registering each cell type once.